doc/source/object-store.rst

```diff
@@ -0,0 +1,169 @@
+Object Store (Plasma)
+=====================
+
+Plasma is an in-memory object store that is being developed as part of `Apache Arrow`_. Ray uses Plasma to efficiently transfer objects across different processes and different nodes.
+
+All objects in Plasma object store are **immutable** and held in shared memory. This is so that they can be accessed efficiently by many workers on the same node.
+
+Note that Plasma does not support all data-types, and in certain cases it may be necessary to either write your own serialization protocol or use Actors to hold objects and transfer object state (i.e., weight matrices) among Ray workers.
+
+Overview
+--------
+
+Ray will place objects in a node object store in the following situations:
+
+1. Calling ``ray.put``
+
+.. code:: python
+
+    y = 2
+    # This places `2` into the object store.
+    object_id = ray.put(y)
+
+2. The return values of a remote function.
+
+.. code:: python
+
+    @ray.remote
+    def remote_function():
+        return 1
+
+    # This places `1` into the object store.
+    object_id = remote_function.remote()
+
+
+3. Arguments to remote functions (except for simple arguments like ints or
+   floats).
+
+.. code:: python
+
+    @ray.remote
+    def remote_function(y):
+        # Note that inside the remote function, the actual argument is provided.
+        return len(y)
+
+    argument = [1, 2, 3, 4]
```

Comment:
No longer accurate. We put them in the object store if `ray._raylet.` returns False and otherwise we pickle them and inline them in the task spec. In this case, the list will be inlined in the task spec. This happens whenever the object consists of primitive types and/or numpy arrays and in aggregate is smaller than about 10KB ish.

Of course `check_simple_value` is not part of the Ray API.